File: ui/src/app/remove_reset.py
import os
import re

def clean_reset_class(directory):
    count = 0
    # Match className="... reset ..." or className={`... reset ...`}
    # We will just replace word boundary 'reset' followed by space, or preceded by space, inside className
    # Let's just find and replace exact matches we know exist to be extremely safe:
    # 'className="reset ' -> 'className="'
    # 'className={`reset ' -> 'className={`'
    # ' className="reset"' -> ' className=""'
    # ' "reset ' -> ' "' (for nested logic like `${scrolled ? "reset top-link" : ""}`)

    for root, _, files in os.walk(directory):
        for file in files:
            if file.endswith(('.tsx', '.ts', '.jsx', '.js')):
                filepath = os.path.join(root, file)
                with open(filepath, 'r', encoding='utf-8') as f:
                    content = f.read()

                new_content = content.replace('className="reset ', 'className="')
                new_content = new_content.replace('className={`reset ', 'className={`')
                new_content = new_content.replace(' className="reset"', ' className=""')
                new_content = new_content.replace(' "reset ', ' "')
                
                # Check for remaining "reset" inside classNames just in case
                # e.g., <button className="top-link reset"
                new_content = new_content.replace(' reset"', '"')
                new_content = new_content.replace(' reset ', ' ')

                if new_content != content:
                    with open(filepath, 'w', encoding='utf-8') as f:
                        f.write(new_content)
                    count += 1
                    print(f"Updated {filepath}")

    print(f"Total files modified: {count}")

clean_reset_class('c:/Users/hinsh/Desktop/saas-prototype/ui/src/app')

File: ui/src/app/remove_reset.js
const fs = require('fs');
const path = require('path');

function walkDir(dir, callback) {
  fs.readdirSync(dir).forEach(f => {
    let dirPath = path.join(dir, f);
    let isDirectory = fs.statSync(dirPath).isDirectory();
    isDirectory ? walkDir(dirPath, callback) : callback(path.join(dir, f));
  });
}

let count = 0;
walkDir('c:/Users/hinsh/Desktop/saas-prototype/ui/src/app', function(filePath) {
  if (filePath.endsWith('.tsx') || filePath.endsWith('.ts') || filePath.endsWith('.jsx') || filePath.endsWith('.js')) {
    let content = fs.readFileSync(filePath, 'utf8');
    let original = content;
    content = content.replace(/className="/g, 'className="');
    content = content.replace(/className={`/g, 'className={`');
    content = content.replace(/ className=""/g, ' className=""');
    content = content.replace(/ "/g, ' "');
    content = content.replace(/"/g, '"');
    content = content.replace(/ /g, ' ');
    if (content !== original) {
      fs.writeFileSync(filePath, content, 'utf8');
      count++;
      console.log('Updated ' + filePath);
    }
  }
});
console.log('Total files modified: ' + count);
